File: Regression/Multi-Linear-Regression.R
# Mutlti Linear Regression

# Importing Dataset
dataset = read.csv('50_Startups.csv')

# Encoding Categorical Values

dataset$State = factor(x=dataset$State,
                       levels = c('New York', 'California', 'Florida'),
                       labels = c(1,2,3))

# splitting dataset into training set and test set
library(caTools)
set.seed(123)
split = sample.split(dataset$Profit, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# fitting linear regression model for training set
regressor = lm( formula = Profit ~ ., data = training_set)
summary(regressor)

# predicting test set results
y_pred = predict(regressor, newdata = test_set )

#  building the optimal model using backward elimintion method
regressor = lm( formula = Profit ~ R.D.Spend + Administration + Marketing.Spend + State,
                data = training_set)
summary(regressor)

regressor = lm( formula = Profit ~ R.D.Spend ,
                data = training_set)
summary(regressor)

File: Regression/Multi-Linear-Regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Oct 23 16:33:39 2017

@author: adhingra
"""
# Mutlti Linear Regression

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# loading dataset
dataset = pd.read_csv('50_Startups.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,4].values

# encoding the categorical values
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X = LabelEncoder()
X[:,3]=labelencoder_X.fit_transform(X[:,3])
onehotencoder = OneHotEncoder(categorical_features=[3])
X = onehotencoder.fit_transform(X).toarray()

# avoiding the dummy variable trap
X = X[:,1:]

# splitting dataset into training data and test data
from sklearn.cross_validation import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y, test_size = 0.2, random_state = 0)

# fitting multiple linear regression model to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# predicting test set results
y_pred = regressor.predict(X_test)

#  building the optimal model using backword elimination
import statsmodels.formula.api as sm
X = np.append(arr = np.ones((50,1)).astype(int), values = X, axis = 1)
X_opt = X[:, [0,1,2,3,4,5]]
regressor_OLS = sm.OLS(endog = y, exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:, [0,1,3,4,5]]
regressor_OLS = sm.OLS(endog = y, exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:, [0,3,4,5]]
regressor_OLS = sm.OLS(endog = y, exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:, [0,3,5]]
regressor_OLS = sm.OLS(endog = y, exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:, [0,3]]
regressor_OLS = sm.OLS(endog = y, exog = X_opt).fit()
regressor_OLS.summary()



